src/groovy/com/recomdata/security/LdapAuthUserDetailsMapper.groovy

```diff
@@ -134,11 +157,19 @@ public class LdapAuthUserDetailsMapper implements UserDetailsContextMapper {
                 "use a subclass if mapUserToContext() is required.");
     }
 
-    protected String mapPassword(Object passwordValue) {
-        if (!(passwordValue instanceof String)) {
-            passwordValue = new String((byte[]) passwordValue);
+    protected String mapPassword(DirContextOperations ctx) {
+        if (noPassword) {
+            return 'NO_PASSWORD'
+        }
+        String password = ''
+        Object passwordValue = ctx.getObjectAttribute("userPasswordRaw");
+        if (passwordValue != null) {
+            if (!(passwordValue instanceof String)) {
+                passwordValue = new String((byte[]) passwordValue);
+            }
+            password = (String) passwordValue
         }
-        return (String) passwordValue;
+        return springSecurityService.encodePassword(password)
```

Comment:
I don't think we want to encourage people to store unencrypted passwords on their ldap directory. And even if we did want, this shouldn't be enabled by default (`noPassword` defaults to false -- btw, double negation is unnecessarily confusing; something like `createPassword` would be clearer) and the name of the attribute should arguably be configurable. I'm also concerned about the default case here, which is that passwordValue is null and noPassword is false; in this case `SpringSecurityService::encodePassword` will be called with `null` as an argument.